### PR TITLE
Add system tests for Central Management

### PR DESCRIPTION
**Pull request by exekias**

### CODE REVIEW

**Comment by exekias** [comment]
WIP, want to check if this actually launches tests

**Comment by ruflin** [comment]
@sayden If this approach works, I think we should copy it also over to x-pack metricbeat.

**Comment by exekias** [comment]
Testing ma[g|k]files in place and working, will add tests now

**Comment by ruflin** [comment]
Change LGTM. I wonder if we should the base infrastructure already in place with a fake test?

**Comment by exekias** [comment]
Broken test is failing as expected: https://travis-ci.org/elastic/beats/jobs/448269164

File: x-pack/libbeat/docker-compose.yml [hunk 1]
```diff
@@ -27,3 +28,21 @@ services:
...
       service: elasticsearch
+    healthcheck:
+      test: ["CMD", "curl", "-u", "elastic:changeme", "-f", "http://localhost:9200"]
+    environment:
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+      - "network.host="
+      - "transport.host=127.0.0.1"
+      - "http.host=0.0.0.0"
+      - "xpack.security.enabled=true"
+      - "xpack.license.self_generated.type=trial"
```
Comment (ruflin) [inline x-pack/libbeat/docker-compose.yml]: @exekias Interesting, didn't know such a config exist. We should probably also use this for the ML tests as there we trigger it through the API.

Comment (exekias) [reply x-pack/libbeat/docker-compose.yml]: yes, that would still enable it for all tests in that scenario, I'm wondering if that's what we want, or if it has some side effects

File: libbeat/scripts/Makefile [hunk 2]
```diff
@@ -213,7 +213,7 @@ system-tests-environment: prepare-tests build-image
 	${DOCKER_COMPOSE} run -e INTEGRATION_TESTS=1 -e TESTING_ENVIRONMENT=${TESTING_ENVIRONMENT} -e DOCKER_COMPOSE_PROJECT_NAME=${DOCKER_COMPOSE_PROJECT_NAME} beat make system-tests
 	#This is a hack to run x-pack/filebeat module tests
 	@XPACKBEAT="${ES_BEATS}/x-pack/${BEAT_NAME}" ; \
-	if [ -e "$$XPACKBEAT/tests/system" ] && [ $(XPACK_ONLY) = false ]; then \
+	if [ -e "$$XPACKBEAT/tests/system" ] && [ "$(BEAT_NAME)" != "libbeat" ] && [ $(XPACK_ONLY) = false ]; then \
```
Comment (exekias) [inline libbeat/scripts/Makefile]: @andrewkroh FYI, I added this exception here

Comment (andrewkroh) [reply libbeat/scripts/Makefile]: Thanks for letting me know. I am working on a way to directly invoke the tests via `mage integTest` which well let us eventually clean up the Make logic.

**Comment by exekias** [comment]
Tests seem to work locally but fail on travis, I'm currently investigating this, seems related to kibana authentication

File: libbeat/tests/system/beat/beat.py [hunk 3]
```diff
@@ -46,8 +46,7 @@ def __init__(self, args, outputfile, env={}):
 
     def start(self):
         # ensure that the environment is inherited to the subprocess.
-        variables = os.environ.copy()
-        variables = variables.update(self.env)
```
Comment (exekias) [inline libbeat/tests/system/beat/beat.py]: this was wrong, `dict.update` returns nothing

File: libbeat/tests/system/beat/beat.py [hunk 4]
```diff
@@ -46,8 +46,7 @@ def __init__(self, args, outputfile, env={}):
...
+        variables = os.environ.copy().update(self.env)
```
Comment (exekias) [inline libbeat/tests/system/beat/beat.py]: this was wrong, `dict.update` returns nothing

File: libbeat/tests/system/beat/beat.py [hunk 5]
```diff
@@ -47,7 +47,7 @@ def __init__(self, args, outputfile, env={}):
     def start(self):
         # ensure that the environment is inherited to the subprocess.
         variables = os.environ.copy()
-        variables = variables.update(self.env)
+        variables.update(self.env)
```
Comment (exekias) [inline libbeat/tests/system/beat/beat.py]: this was wrong, `dict.update` returns nothing

Comment (ph) [reply libbeat/tests/system/beat/beat.py]: You are correct, I am really to used to ruby that always returned the modified values by default.

I've created https://github.com/elastic/beats/pull/8998 with your change, It need to go in 6.4 6.5, 6.x

File: libbeat/scripts/Makefile (hunk 2 above)
Comment (andrewkroh) [reply libbeat/scripts/Makefile]: That PR is 
https://github.com/elastic/beats/pull/8956. FYI

File: x-pack/libbeat/tests/system/test_management.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+        # Settings backup has been created
+        assert os.path.isfile(os.path.join(
+            self.working_dir, "mockbeat.yml.bak"))
+        backup_content = open(config_path + ".bak", 'r').read()
+        assert config_content == backup_content
+
+    def test_enroll_bad_pw(self):
+        """
+        Enroll the beat in Kibana Central Management
```
Comment (andrewkroh) [inline x-pack/libbeat/tests/system/test_management.py]: Description is the same as previous test.

**Pull request merged**